Source/FieldSolver/ElectrostaticSolvers/ElectrostaticSolver.H

```diff
@@ -0,0 +1,163 @@
+/* Copyright 2024 The WarpX Community
+ *
+ * This file is part of WarpX.
+ *
+ * Authors: Roelof Groenewald, Arianna Formenti, Revathi Jambunathan
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef WARPX_ELECTROSTATICSOLVER_H_
+#define WARPX_ELECTROSTATICSOLVER_H_
+
+#include "PoissonBoundaryHandler.H"
+#include "Fluids/MultiFluidContainer.H"
+#include "Particles/MultiParticleContainer.H"
+#include "Utils/WarpXProfilerWrapper.H"
+#include "WarpX.H"
+
+#include <AMReX_Array.H>
+
+
+/**
+ * \brief Base class for Electrostatic Solver
+ *
+ */
+class ElectrostaticSolver
+{
+public:
+    ElectrostaticSolver() = default;
+    ElectrostaticSolver( int nlevs_max );
+
+    virtual ~ElectrostaticSolver();
+
+    // Prohibit Move and Copy operations
+    ElectrostaticSolver(const ElectrostaticSolver&) = delete;
+    ElectrostaticSolver& operator=(const ElectrostaticSolver&) = delete;
+    ElectrostaticSolver(ElectrostaticSolver&&) = delete;
+    ElectrostaticSolver& operator=(ElectrostaticSolver&&) = delete;
+
+    void ReadParameters ();
+
+    virtual void InitData () {}
+
+    /**
+     * \brief Computes charge density, rho, and solves Poisson's equation
+     *        to obtain the associated electrostatic potential, phi.
+     *        Using the electrostatic potential, the electric field is computed
+     *        in lab frame, and if relativistic, then the electric and magnetic
+     *        fields are computed using potential, phi, and
+     *        velocity of source for potential, beta.
+     *        This function must be defined in the derived classes.
```

Comment:
```suggestion
     * \brief Modify the E and B fields to add the space-charge fields
     *           This is done by solving the Poisson equation, using the `phi` multiFab
```

Reply:
discussed offline. Remi will add this commit in PR #5230